### PR TITLE
51492 Webhook endpoint for subscription to Enterprise Event Service for File Share Service New Files Published event

### PR DESCRIPTION
**Pull request by Avinash15515**
MSI Webhook endpoint for EES Integration.

IsFssCacheEnabled flag is set to false by default

UI changes in the following pages (Reverted):- (corrected alignment)
1. Cumulative tab

### CODE REVIEW

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/WebhookController.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    public class WebhookController : BaseController<WebhookController>
+    {
+        public WebhookController(IHttpContextAccessor contextAccessor, ILogger<WebhookController> logger) : base(contextAccessor, logger)
+        {
+        }
+
+        [HttpOptions]
+        [Route("/webhook/newfilespublished")]
+        public IActionResult NewFilesPublishedOptions()
+        {
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/WebhookController.cs:17]: Please add unit test cases for both options and webhook.

**Review by usha13529 — CHANGES_REQUESTED** [review]

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/WebhookController.cs (hunk 1 above)
Comment (Avinash15515) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/WebhookController.cs:17]: Test cases added

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/WebhookControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+        [Test]
+        public void WhenValidHeaderRequestedInNewFilesPublishedOptions_ThenReturnsOkResponse()
+        {
+            _controller.ControllerContext.HttpContext = new DefaultHttpContext();
+            _fakeContextAccessor.HttpContext.Request.Headers.Add("WebHook-Request-Origin", "test.example.com");
+
+            var result = (OkObjectResult)_controller.NewFilesPublishedOptions();
+
+            Assert.AreEqual(200, result.StatusCode);
```
Comment (usha13529) [inline Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/WebhookControllerTests.cs:40]: Please validate response header contains same header values received in Options request.

**Review by usha13529 — CHANGES_REQUESTED** [review]

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/WebhookControllerTests.cs (hunk 2 above)
Comment (Avinash15515) [reply Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/WebhookControllerTests.cs:49]: Added additional checks

**Pull request merged**